Source/WTF/wtf/text/StringParsingBuffer.h

```diff
@@ -55,8 +53,8 @@ class StringParsingBuffer final {
 
     constexpr void setPosition(const CharacterType* position)
     {
-        ASSERT(position <= m_data.data() + m_data.size());
-        m_data = { position, m_data.data() + m_data.size() };
+        ASSERT(position <= std::to_address(m_data.end()));
+        m_data = unsafeMakeSpan(position, std::to_address(m_data.end()) - position);
```

Comment:
This operation is truly not-bounds-safe, even based on outside-the-language human local reasoning. It would be nice to express this in a checkable way.

Here's one option:

m_data = m_data.subspan(position - m_data.data());

(The subtraction may underflow, but the new position will still be bounds-checked.)

Another option is to change the argument type to be an offset.

Reply:
Your solution doesn't work. Caller often use `setPosition()` to rewind to an address *before* the span. The long term solution is to rewrite call sites somehow to not rely on this.